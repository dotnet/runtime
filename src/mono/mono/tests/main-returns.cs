
using System;
using System.Threading;

public class foo {
	public static void Main() {
		// it will actually return 0 only if the thread finishes executing
		Environment.ExitCode = 1;
		Thread thr=new Thread(new ThreadStart(foo.thread));
		thr.Start();
		Thread.Sleep(1200);
		Console.WriteLine("Main thread returns");
	}

	public static void thread() {
		Console.WriteLine("Thread running");
		Thread.Sleep(500);
		Console.WriteLine("Thread running");
		Thread.Sleep(500);
		Console.WriteLine("Thread running");
		Thread.Sleep(500);
		Console.WriteLine("Thread running");
		Thread.Sleep(500);
		Console.WriteLine("Thread running");
		Environment.ExitCode = 0;
	}
}

