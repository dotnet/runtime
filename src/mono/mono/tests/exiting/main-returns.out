Thread running
Thread running
Thread running
Main thread returns
Thread running
Thread running
